.github/workflows/macos.yml

```diff
@@ -0,0 +1,260 @@
+      FEATURE_TOOLS_CTS: ${{ matrix.options.tools_cts }}
+      PACKAGE: ${{ matrix.options.package && matrix.options.package || 'NO' }}
+      SUPPORT_OPENCL: ${{ matrix.options.opencl }}
+      SUPPORT_SSE: ${{ matrix.options.sse }}
+      WASM_BUILD: ${{ matrix.options.wasm && matrix.options.wasm || 'OFF' }}
+      WERROR: ${{ matrix.options.werror && matrix.options.werror || 'ON' }}
+      # CC Handled by job step
+      # CXX Handled by job step
+
+      PYTHON_DIST_DIR: interface/python_binding/dist
+
+      VULKAN_SDK_VER: ${{ matrix.vk_sdk_ver }}
+    # This is just to tell the Vulkan install script where to install.
+    # NOTE: Hardcoded $HOME expansion, due to 'env' context being unavailble in env:
+      VULKAN_INSTALL_DIR: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}
+      VULKAN_SDK: /Users/runner/VulkanSDK/${{ matrix.vk_sdk_ver }}/macOS
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Fetch all history to make sure tags are
+          # included (used for version creation)
+          fetch-depth: 0
+
+      - name: before_install
+        run: |
+          echo -n "Running on the following GitHub Actions CI runner: " && uname -a
+          echo -n "CMake version on the runner is " && cmake --version
+          echo -e "machine github.com\n  login $GITHUB_TOKEN" >> ~/.netrc # Prevent rate limiting on Git LFS.
+          cat ~/.netrc
+          if [[ -n "$FEATURE_LOADTESTS" && "$FEATURE_LOADTESTS" != "OFF" ]]; then
+            # Install vcpkg.
+            pushd $HOME
+            git clone https://github.com/microsoft/vcpkg
+            cd vcpkg && ./bootstrap-vcpkg.sh -disableMetrics && echo "VCPKG_ROOT=`pwd`" >> $GITHUB_ENV
+            popd
+          fi
+
+      - name: Install Doxygen
+        if: matrix.options.doc
+        uses: ssciwr/doxygen-install@v1
+
+      - name: Install Graphviz
+        if: matrix.options.doc
+        uses: ts-graphviz/setup-graphviz@v2
+
+      - name: Run install script
+        run: ./scripts/install_macos.sh
+
+      - name: Set up Python 3.11
+        if: matrix.options.py == 'ON'
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11.4'
+
+      - name: before_script
+        run: |
+          ./scripts/before_build_macos.sh
+          if [ "$FEATURE_TOOLS_CTS" = "ON" ]; then
+            git submodule update --init --recursive tests/cts
+          fi
+          # Make sure embedded dates are correct.
+          ./install-gitconfig.sh
+          scripts/smudge_date.sh
+
+      - name: script
+        run: |
+          set -e # Instead of trying to chain everything together with &&
+                 # otherwise subsequent commands will swallow bad exit status.
+          if [ "$PLATFORM" = "macOS" ]; then
+            ./scripts/build_macos.sh
+          else
+            ./scripts/build_ios.sh
+          fi
+          set +e
+
+      - name: Notarize
+        if: matrix.options.platform == 'macOS'
```

Comment:
Only run this step if secrets are available, e.g APPLE_ID or DEVELOPMENT_TEAM is set. PR's from forks, where secrets are not available, must still run.

Reply:
Done.